libbeat/beat/beat.go

```diff
@@ -141,43 +161,43 @@ func (b *Beat) Start() error {
 
 // Reads and parses the default command line params
 // To set additional cmd line args use the beat.CmdLine type before calling the function
-func (beat *Beat) CommandLineSetup() {
+// The second return param is to detect if system should exit. True if should exit
+// Exit can also be without error
+func (beat *Beat) CommandLineSetup() (error, bool) {
 
 	// The -c flag is treated separately because it needs the Beat name
 	err := cfgfile.ChangeDefaultCfgfileFlag(beat.Name)
 	if err != nil {
-		fmt.Printf("Failed to fix the -c flag: %v\n", err)
-		os.Exit(1)
+		return fmt.Errorf("Failed to fix the -c flag: %v\n", err), true
 	}
 
 	flag.Parse()
 
 	if *printVersion {
 		fmt.Printf("%s version %s (%s)\n", beat.Name, beat.Version, runtime.GOARCH)
-		os.Exit(0)
+		return nil, true
 	}
 
 	// if beater implements CLIFlags for additional CLI handling, call it now
 	if flagsHandler, ok := beat.BT.(FlagsHandler); ok {
 		flagsHandler.HandleFlags(beat)
 	}
+
+	return nil, false
 }
 
 // LoadConfig inits the config file and reads the default config information
 // into Beat.Config. It exists the processes in case of errors.
-func (b *Beat) LoadConfig() {
+func (b *Beat) LoadConfig() error {
 
 	err := cfgfile.Read(&b.Config, "")
 	if err != nil {
-		// logging not yet initialized, so using fmt.Printf
-		fmt.Printf("Loading config file error: %v\n", err)
-		os.Exit(1)
+		return fmt.Errorf("Loading config file error: %v\n", err)
```

Comment:
by convention error strings should start with lowercase, as these are often embedded in some other message.